art/experiment/Experiment.py

```diff
@@ -73,27 +73,43 @@ def check_if_must_be_run(self, step: "Step", checks: List[Check]):
             )
             return True
 
-        print(f"Step {step.name}_{step.get_step_id()} was already completed.")
+        # print(f"Step {step.name}_{step.get_step_id()} was already completed.")
         return False
 
     def run_all(self):
+        steps_status = {step["step"].get_full_step_name(): ("Not run", None) for step in self.steps}
         for step in self.steps:
             self.metric_calculator.compile(step["skipped_metrics"])
             step, checks = step["step"], step["checks"]
 
             self.state.current_step = step
 
             if not self.check_if_must_be_run(step, checks):
+                steps_status[step.get_full_step_name()] = ("Skipped", step.get_results())
                 self.fill_step_states(step)
                 continue
 
             step.add_result("hash", step.get_hash())
+            
+            try:
+                step(self.state.step_states, self.datamodule, self.metric_calculator)
+                self.check_checks(step, checks)
+                steps_status[step.get_full_step_name()] = ("Completed", step.get_results())
+            except Exception as e:
+                steps_status[step.get_full_step_name()] = ("Failed", step.get_results())
+                print(f"Step {step.get_full_step_name()} failed with error: {str(e)}")
+                continue
 
-            step(self.state.step_states, self.datamodule, self.metric_calculator)
-
-            self.check_checks(step, checks)
             self.fill_step_states(step)
 
+        print("\n\nSteps status:")
+        for step_name, status in steps_status.items():
```

Comment:
This is next reason to use dataclasses or just a class. If you implemented __repr__ magic method you could omit this loop

and do: print(status) if you make status an object (dataclass)